### PR TITLE
rm: ensure context builder is not removed with all-inactive flag

### PR DESCRIPTION
**Pull request by crazy-max**
forgot to check if a context builder is being removed when using `all-inactive` flag (can happen with remote context).

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

**Comment by jedevc** [comment]
It's possible we could break this down a bit differently, there's some already with the `rmAllInactive` method. We could potentially refactor to get a list of all inactive builders, and then `rm` them all, so that we only have one set of context checking steps.

**Comment by crazy-max** [comment]
Was thinking doing that in a follow-up with another refactoring about builder nodes.

File: commands/rm.go [hunk 1]
```diff
@@ -149,13 +149,23 @@ func rmAllInactive(ctx context.Context, txn *store.Txn, dockerCli command.Cli, i
...
 	eg, _ := errgroup.WithContext(ctx)
 	for _, b := range builders {
 		func(b *nginfo) {
 			eg.Go(func() error {
 				if err := loadNodeGroupData(ctx, dockerCli, b); err != nil {
 					return errors.Wrapf(err, "cannot load %s", b.ng.Name)
 				}
+				for _, cb := range ctxbuilders {
+					if b.ng.Driver == "docker" && len(b.ng.Nodes) == 1 && b.ng.Nodes[0].Endpoint == cb.Name {
+						return errors.Errorf("context builder cannot be removed, run `docker context rm %s` to remove this context", cb.Name)
```
Comment (jedevc) [inline commands/rm.go:166]: I wonder if we should just `return` to skip over the context builders here, instead of returning an error.

Comment (crazy-max) [reply commands/rm.go:166]: I see what you mean but I think the user would expect this builder to be removed and if we don't return an error he will be confused imo.

**Comment by crazy-max** [comment]
As discussed with @tonistiigi this is not needed as `txn.List()` only returns instances from buildx store and not context ones.

**Pull request closed without merging**